src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -320,16 +498,12 @@ private static IEnumerable<ProjectItem> GetPackageReferences(Project project, Li
 
             foreach (var framework in frameworks)
             {
-                var globalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
-                { { "TargetFramework", framework } };
-                var projectPerFramework = GetProject(project.FullPath, globalProperties);
-
-                mergedPackageReferences.AddRange(GetPackageReferences(projectPerFramework, libraryDependency));
-                ProjectCollection.GlobalProjectCollection.UnloadProject(projectPerFramework);
```

Comment:
Instead of doing the project loading here, I do all of it (with the unloading) in a different function that does exactly the same thing, so it should be fine